low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -314,11 +324,12 @@ def validate(self):
             raise SetupError(1, msg)
         logger.info("%s - Validation done" % self.name)
 
-        if self.args.coverage and 'self' in self.args.plan[0]:
-            raise SetupError(errno.EINVAL,
-                             "%s - Argument validation failure. %s",
-                             self.name,
-                             "Code coverage can not be enabled with self tests.")
+        if self.args.coverage and self.args.plan[0] in TEST_REQ_SERVICE_RESTART:
```

Comment:
please add a code comment here so later we know the reason without thinking about it